CMakeFilters.cmake

```diff
@@ -59,9 +59,9 @@ if (HDF5_ENABLE_Z_LIB_SUPPORT)
       find_package (ZLIB NAMES ${ZLIB_PACKAGE_NAME}${HDF_PACKAGE_EXT} COMPONENTS static shared)
       if (NOT ZLIB_FOUND)
         find_package (ZLIB) # Legacy find
-        if (ZLIB_FOUND)
-          set (LINK_COMP_LIBS ${LINK_COMP_LIBS} ${ZLIB_LIBRARIES})
-        endif ()
+      endif ()
+      if (ZLIB_FOUND)
+        set (LINK_COMP_LIBS ${LINK_COMP_LIBS} ${ZLIB_LIBRARIES})
```

Comment:
This line should just be added to the if block that is next. (line 67)

Reply:
Look at line 94.  I think this set (LINK_COMP_LIBS line should replace the one at line 94 as that is the end of any ZLIB_FOUND checking and ZLIB_LIBRARIES is set correctly no matter which path was taken.